README.md

```diff
@@ -5,40 +5,191 @@ Please edit this README.md file with your answers to the worksheet questions.
+|true |true |true |true |true                  |true          |
 
 ### d
+|A    |B    |C    |D    |(A OR B) AND (A OR C)|A OR (B AND C)|
+|-----|-----|-----|-----|---------------------|--------------|
+|false|false|false|false|false                |false         |
+|false|false|false|true |false                |false         |
+|false|false|true |false|false                |false         |
+|false|false|true |true |false                |false         |
+|false|true |false|false|false                |false         |
+|false|true |false|true |false                |false         |
+|false|true |true |false|true                 |true          |
+|false|true |true |true |true                 |true          |
+|true |false|false|false|true                 |true          |
+|true |false|false|true |true                 |true          |
+|true |false|true |false|true                 |true          |
+|true |false|true |true |true                 |true          |
+|true |true |false|false|true                 |true          |
+|true |true |false|true |true                 |true          |
+|true |true |true |false|true                 |true          |
+|true |true |true |true |true                 |true          |
 
 ## Question 4
 
 ### a
+|file_exists("a.txt")|file_exists("b.txt")|first program|second program|
+|--------------------|--------------------|-------------|--------------|
+|false               |false               |true         |true          |
+|false               |true                |true         |true          |
+|true                |false               |true         |true          |
+|true                |true                |false        |false         |
 
 ### b
+|type(x) == int|x > 7|type(x) == float|first program|second program|
+|--------------|-----|----------------|-------------|--------------|
+|false         |false|false           |false        |false         |
+|false         |false|true            |false        |false         |
+|false         |true |false           |false        |false         |
+|false         |true |true            |true         |true          |
+|true          |false|false           |false        |false         |
+|true          |false|true            |false        |false         |
+|true          |true |false           |true         |true          |
+|true          |true |true            |true         |true          |
 
 ### c
-
+|X == 0 |Y == 0|first program|second program|
+|-------|------|-------------|--------------|
+|true   |true  |true         |false         |
+|true   |false |false        |true          |
+|false  |true  |false        |true          |
+|false  |false |false        |true          |
 ### d
-
+|X > 10 |Y > 0 |first program|second program|
+|-------|------|-------------|--------------|
+|false  |false |true         |false         |
+|false  |true  |true         |false         |
+|true   |false |false        |false         |
+|true   |true  |true         |true          |
```

Comment:
This doesn't account for the `x > 0` condition which also appears in these programs?